gpiozero/pins/native.py

```diff
@@ -219,9 +219,10 @@ def __new__(cls, number):
             try:
                 cls.PI_INFO.physical_pin('GPIO%d' % number)
             except PinNoPins:
-                warnings.warn(
-                    PinNonPhysical(
-                        'no physical pins exist for GPIO%d' % number))
+                if PinNonPhysical.printWarnings:
+                    warnings.warn(
+                        PinNonPhysical(
+                            'no physical pins exist for GPIO%d' % number))
```

Comment:
It's just a gut feeling, but having `PinNonPhysical` mentioned here twice feels like an anti-pattern.
I wonder if it might be better to wrap `warnings.warn` inside another function, and have that function check the `.printWarnings` class attribute of the passed-in warning?

@waveform80 will probably have a better idea about the 'most Pythonic' solution ;)